plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/AllFilesTable.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.spi.Page;
+import io.trino.spi.connector.ConnectorPageSource;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.ConnectorTableMetadata;
+import io.trino.spi.connector.ConnectorTransactionHandle;
+import io.trino.spi.connector.FixedPageSource;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.predicate.TupleDomain;
+import io.trino.spi.type.TypeManager;
+import org.apache.iceberg.Table;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public class AllFilesTable
+        extends AbstractFilesTable
+{
+    private final ConnectorTableMetadata tableMetadata;
+    private final Table icebergTable;
+
+    public AllFilesTable(SchemaTableName tableName, TypeManager typeManager, Table icebergTable)
+    {
+        this.icebergTable = requireNonNull(icebergTable, "icebergTable is null");
+        this.tableMetadata = tableMetadata(tableName, typeManager);
+    }
+
+    @Override
+    public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
+    {
+        return new FixedPageSource(buildPages());
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return tableMetadata;
+    }
+
+    private List<Page> buildPages()
+    {
+        ImmutableList.Builder<Page> pages = ImmutableList.builder();
+        icebergTable.snapshots().forEach(snapshot -> pages.addAll(buildPages(icebergTable, snapshot.snapshotId())));
```

Comment:
Say we insert in `v1` of the table a file `foobar.parquet`.
Does this mean that in `vn` of the table we'll have `n` occurences of `foobar.parquet` ?

Can you check that this functionality if the same as in Spark?

Reply:
files will repeat as the old snapshots will be indirectly referring to those files through the manifests files.

Reply:
Could you please add a corresponding test into `io.trino.tests.product.iceberg.TestIcebergSparkCompatibility`  to ensure that the output of the tables is the same?

Reply:
`trino`

```
create table iceberg.default.commontable (id integer);

insert into iceberg.default.commontable values (1);
insert into iceberg.default.commontable values (2);
insert into iceberg.default.commontable values (3);

select * from iceberg.default."commontable$all_files" order by file_path;
-- 6 entries : 3 corresponding to the file containing value `1` , 2 corresponding to the file containing value `2`, 1 corresponding to the file containing value `3`
```

`spark-iceberg`

```
create table test1 (x int) using iceberg tblproperties('write.format.default'= 'PARQUET', 'format-version'=2);
insert into test1 values (1);
insert into test1 values (2);
insert into test1 values (3);

select * from test1.all_data_files;
-- 3 files -> 1 for each insert
```

Reply:
@findinpath looks like Spark is deducing the manifests in [AllDataFiles](https://github.com/apache/iceberg/blob/718fc5ae9651d016dcfd2cb80d57bfb7366e984b/core/src/main/java/org/apache/iceberg/BaseAllMetadataTableScan.java#L69),  which avoids these dupes. where we are iterating through the snapshots.

I did bit more digging of the Spark code, and in local able to use the Iceberg API like `MetadataUtils.createMetadataTableInstance` to make it compatible with the Spark and Trino.
We discussed this earlier as well, this will also enable use to be in intact with the Iceberg APIs/upgrades and bug fixes as well.

I will do bit more testing and then update the PR.

thanks.